src/main/webapp/src/app/map/map.component.ts

```diff
@@ -11,66 +11,89 @@ import { HiddenGemService } from '../hidden-gem.service';
 })
 
 export class MapComponent implements AfterViewInit {
-
+  
   @ViewChild("mapContainer", { static: false })
   mapContainer!: ElementRef;
   map!: google.maps.Map;
+  hiddenGems = this.appComponent.hiddenGems;
+  type!: string;
   location = {} as Location;
 
-  constructor(private locationService: LocationService) {
-      this.locationService.getLocation
+  constructor( private appComponent: AppComponent, private locationService: LocationService) {
+    this.locationService.getLocation
       .subscribe(location => {
         this.location = location;
-    })
+      })
   }
 
+  ngDoCheck() {
+    if (this.hiddenGems !== this.appComponent.hiddenGems) {
+      this.hiddenGems = this.appComponent.hiddenGems;
+      this.mapInitializer();
+    }
+  }
+  
   ngAfterViewInit(): void {
```

Comment:
Nit - You can also remove this function

Reply:
ngAfterViewInit?

Reply:
Yes, the function is empty so I think you should be able to remove it? And you will need to remove the "implements AfterViewInit" at the top as well

Reply:
ah ok removed